File: backend/InfluencerAPI/Models/Deal.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace InfluencerAPI.Models;

public class Deal
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    [BsonElement("dealNumber")]
    public int DealNumber { get; set; } // ID legible: DEAL-001

    // === Información General ===
    [BsonElement("creatorName")]
    public string CreatorName { get; set; } = string.Empty;

    [BsonElement("clientName")]
    public string ClientName { get; set; } = string.Empty;

    [BsonElement("campaignName")]
    public string CampaignName { get; set; } = string.Empty;

    [BsonElement("contentType")]
    public string ContentType { get; set; } = string.Empty; // TikTokPost | InstagramReel | InstagramStory | YouTubeVideo

    [BsonElement("currency")]
    public string Currency { get; set; } = "COP"; // COP | USD

    [BsonElement("totalValue")]
    public decimal TotalValue { get; set; }

    [BsonElement("creatorPayment")]
    public decimal CreatorPayment { get; set; } // 80% calculado

    [BsonElement("commission")]
    public decimal Commission { get; set; } // 20% calculado

    // === Estado ===
    [BsonElement("status")]
    public string Status { get; set; } = "Confirmado"; // Confirmado | Publicado | Cancelado

    // === Publicación ===
    [BsonElement("publicationLink")]
    public string? PublicationLink { get; set; }

    [BsonElement("publicationDate")]
    public DateTime? PublicationDate { get; set; }

    [BsonElement("notes")]
    public string? Notes { get; set; }

    // === Aprobación para Facturación ===
    [BsonElement("approvedToBill")]
    public bool ApprovedToBill { get; set; } = false;

    [BsonElement("approvedToBillDate")]
    public DateTime? ApprovedToBillDate { get; set; }

    // === Control de Pagos ===
    [BsonElement("creatorPaymentReceived")]
    public bool CreatorPaymentReceived { get; set; } = false;

    [BsonElement("creatorPaymentDate")]
    public DateTime? CreatorPaymentDate { get; set; }

    [BsonElement("commissionReceived")]
    public bool CommissionReceived { get; set; } = false;

    [BsonElement("commissionReceivedDate")]
    public DateTime? CommissionReceivedDate { get; set; }

    // === Auditoría ===
    [BsonElement("createdBy")]
    public string CreatedBy { get; set; } = string.Empty;

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: backend-node/models/Deal.js
const mongoose = require('mongoose');

const dealSchema = new mongoose.Schema(
  {
    dealNumber: { type: Number, required: true },

    creatorName: { type: String, required: true },
    clientName: { type: String, required: true },
    campaignName: { type: String, required: true },
    contentType: { type: String, required: true },
    currency: { type: String, default: 'COP' },
    totalValue: { type: Number, required: true },
    creatorPayment: { type: Number, required: true },
    commission: { type: Number, required: true },

    status: { type: String, default: 'Confirmado' },

    publicationLink: { type: String, default: null },
    publicationDate: { type: Date, default: null },
    notes: { type: String, default: null },

    approvedToBill: { type: Boolean, default: false },
    approvedToBillDate: { type: Date, default: null },

    creatorPaymentReceived: { type: Boolean, default: false },
    creatorPaymentDate: { type: Date, default: null },
    commissionReceived: { type: Boolean, default: false },
    commissionReceivedDate: { type: Date, default: null },

    createdBy: { type: String, required: true },
  },
  { timestamps: { createdAt: 'createdAt', updatedAt: 'updatedAt' }, collection: 'deals' }
);

module.exports = mongoose.models.Deal || mongoose.model('Deal', dealSchema);
